library/std/src/io/mod.rs

```diff
@@ -1491,6 +1491,31 @@ impl<'a> IoSlice<'a> {
             bufs[0].advance(left);
         }
     }
+
+    /// Get the underlying bytes as a rust slice with the original lifetime
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// #![feature(io_slice_as_bytes)]
+    /// use std::io::IoSlice;
+    ///
+    /// let data = b"abcdef";
+    ///
+    /// let mut io_slice = IoSlice::new(data);
+    /// let tail = &io_slice.into_bytes()[3..];
+    ///
+    /// // This works because `tail` doesn't borrow `io_slice`
+    /// io_slice = IoSlice::new(tail);
+    ///
+    /// assert_eq!(io_slice.into_bytes(), b"def");
+    /// ```
+    #[unstable(feature = "io_slice_as_bytes", issue = "111277")]
```

Comment:
The issue number points to this PR; you should open a tracking issue and link it instead.

Also feature gate `io_slice_as_bytes` -> `io_slice_into_bytes` if that is the name